### PR TITLE
Feature/jwt auth

### PR DESCRIPTION
**Pull request by iamkahvi**
- Adding auth router group
- Adding /auth route
- Adding email package using mailgun
- Adding /auth route handlers
- Adding isAuthorized middleware to verify JWT
- Adding generateJWT function
- Replacing the prints with logs

### CODE REVIEW

**Comment by iamkahvi** [comment]
@jayden-chan

File: main.go [hunk 1]
```diff
@@ -52,14 +113,118 @@ func pong(c *gin.Context) {
...
 func fetchUser(d *storage.DB) gin.HandlerFunc {
 	return func(c *gin.Context) {
 		d.Number++
 		id := c.Param("id")
 
 		val, err := strconv.Atoi(id)
 		if err != nil {
-			fmt.Println("Invalid format")
+			log.Println("Invalid format")
 			c.Header("Access-Control-Allow-Origin", "*")
```
Comment (jayden-chan) [inline main.go]: This code duplication is really nasty. Not sure if that's just Go or if there's something you can do to fix that

File: storage/storage.go [hunk 2]
```diff
@@ -30,10 +30,11 @@ type User struct {
 
 // DB : The database struct to store stuff from the server
 type DB struct {
-	User   User
-	Item   Item
-	Number int
-	db     *gorm.DB
+	User    User
+	Item    Item
+	Number  int
```
Comment (jayden-chan) [inline storage/storage.go:35]: What's Number field for?

File: main.go [hunk 3]
```diff
@@ -52,14 +113,118 @@ func pong(c *gin.Context) {
...
+		return 0, err
+	}
+
+	return int(r.Int64()), nil
+}
+
+func generateRandBytes() ([]byte, error) {
+	c := 10
+	b := make([]byte, c)
+	_, err := rand.Read(b)
```
Comment (jayden-chan) [inline main.go:143]: Where did you find this method for generating random bytes? Seems incredibly dubious, the bytes here are probably undefined not random. This is a security concern

**Review by jayden-chan — COMMENTED** [review]
Didn't really have a chance to look at all of it but it seems ok to me. Why did you choose Go for this btw?

File: main.go (hunk 1 above)
Comment (iamkahvi) [reply main.go]: True, I should probably have a universal function for that. Not sure what the best practices are

File: storage/storage.go (hunk 2 above)
Comment (iamkahvi) [reply storage/storage.go:35]: Something I had in there to track the number of requests, not relevant tho

File: main.go (hunk 3 above)
Comment (iamkahvi) [reply main.go:143]: That's from [here](https://golang.org/pkg/crypto/rand/#example_Read), but I don't need that function anyway so I should get rid of it

**Comment by iamkahvi** [comment]
Just wanted to learn Go, I'm enjoying it so far. And using gin is cool

**Comment by iamkahvi** [comment]
> Didn't really have a chance to look at all of it but it seems ok to me. Why did you choose Go for this btw?

Just wanted to learn Go, I'm enjoying it so far. And using gin is cool

File: main.go (hunk 3 above)
Comment (jayden-chan) [reply main.go:143]: Oh yeah that actually looks fine I just misread the code

**Pull request merged**